compiler warning: double definition of macro JSON_INTERNAL_CATCH

- version 3.2.0

```cpp
#define JSON_CATCH_USER(exception) if(false)
#include "nlohmann/json.hpp"
```
single_include/nlohmann/json.hpp:211:0: warning: "JSON_INTERNAL_CATCH" redefined
single_include/nlohmann/json.hpp:198:0: first definition of macro "JSON_INTERNAL_CATCH"

--- a/include/nlohmann/detail/macro_scope.hpp
+++ b/include/nlohmann/detail/macro_scope.hpp
@@ -62,6 +62,7 @@
 #if defined(JSON_CATCH_USER)
     #undef JSON_CATCH
     #define JSON_CATCH JSON_CATCH_USER
+    #undef JSON_INTERNAL_CATCH
     #define JSON_INTERNAL_CATCH JSON_CATCH_USER
 #endif
 #if defined(JSON_INTERNAL_CATCH_USER)

--- a/single_include/nlohmann/json.hpp
+++ b/single_include/nlohmann/json.hpp
@@ -177,6 +177,7 @@ using json = basic_json<>;
 #if defined(JSON_CATCH_USER)
     #undef JSON_CATCH
     #define JSON_CATCH JSON_CATCH_USER
+    #undef JSON_INTERNAL_CATCH
     #define JSON_INTERNAL_CATCH JSON_CATCH_USER
 #endif
 #if defined(JSON_INTERNAL_CATCH_USER)

